src/hmem.c

```diff
@@ -143,27 +143,27 @@ ssize_t ofi_copy_to_hmem_iov(const struct iovec *hmem_iov,
 				     OFI_COPY_BUF_TO_IOV);
 }
 
-int ofi_hmem_init(enum fi_hmem_iface iface)
+void ofi_hmem_init(void)
 {
+	enum fi_hmem_iface iface;
 	int ret;
 
-	/* Lockless check to see if iface has been initialized. */
-	if (hmem_ops[iface].initialized)
-		return FI_SUCCESS;
-
 	pthread_mutex_lock(&init_lock);
-	if (hmem_ops[iface].initialized) {
-		pthread_mutex_unlock(&init_lock);
-		return FI_SUCCESS;
-	}
 
-	ret = hmem_ops[iface].init();
-	if (ret == FI_SUCCESS)
-		hmem_ops[iface].initialized = true;
+	for (iface = 0; iface < ARRAY_SIZE(hmem_ops); iface++) {
+		if (hmem_ops[iface].initialized)
```

Comment:
This check shouldn't be needed.  This call should only be made once from some common location.

Reply:
Sounds good.